openmls/src/treesync/node/leaf_node.rs

```diff
@@ -458,26 +462,51 @@ impl LeafNode {
     /// The key pair must be stored in the key store by the caller.
     fn new_with_key(
         encryption_key: EncryptionKey,
-        credential_bundle: &CredentialBundle,
+        credential_with_key: CredentialWithKey,
         leaf_node_source: LeafNodeSource,
         capabilities: Capabilities,
         extensions: Extensions,
-        backend: &impl OpenMlsCryptoProvider,
+        signer: &(impl Signer + ?Sized),
     ) -> Result<Self, LibraryError> {
         let leaf_node_tbs = LeafNodeTbs::new(
             encryption_key,
-            credential_bundle.credential().signature_key().clone(),
-            credential_bundle.credential().clone(),
+            credential_with_key,
             capabilities,
             leaf_node_source,
             extensions,
         )?;
 
         leaf_node_tbs
-            .sign(backend, credential_bundle.signature_private_key())
+            .sign(signer)
             .map_err(|_| LibraryError::custom("Signing failed"))
     }
 
+    /// Generate a fresh leaf node with a fresh encryption key but otherwise
+    /// the same properties as the current leaf node.
+    ///
+    /// The newly generated encryption key pair is stored in the key store.
+    ///
+    /// This function can be used when generating an update. In most other cases
+    /// a leaf node should be generated as part of a new [`KeyPackage`].
+    pub fn updated<KeyStore: OpenMlsKeyStore>(
```

Comment:
Do we actually need this function in the public API? Couldn't we instead call `self_update` and have it generate an updated leaf node for us?